### PR TITLE
TunableOp: Performance Improvement

### PR DESCRIPTION
**Pull request by naromero77amd**
This PR reduces the overhead on the CPU side by eliminating the use of c10::str in creating signatures. Instead we use fmt library. TunableOp overhead on the CPU are reduced by around ~40%. The improvement is most noticeable on small GEMMs. This PR does not contain any bug fixes or new features.

### CODE REVIEW

**Review by jeffdaily — APPROVED** [review]

**Comment by naromero77amd** [comment]
@malfet Appreciate review if you have the cycles.

File: aten/src/ATen/cuda/tunable/GemmCommon.h [hunk 1]
```diff
@@ -81,7 +82,15 @@ struct GemmParams : OpParams {
   }
 
   std::string Signature() const override {
-    return c10::str(transa, transb, "_", m, "_", n, "_", k);
```
Comment (malfet) [inline aten/src/ATen/cuda/tunable/GemmCommon.h:84]: Hmm, have you tried `fmt::format`? It should be as fast as `snprintf`

Comment (naromero77amd) [reply aten/src/ATen/cuda/tunable/GemmCommon.h:84]: Great suggestion to use fmt. Even better performance. Appreciate if you can take a look at the changes to CMake changes.

**Pull request closed without merging**